src/ModelObjects/component.rs

```diff
@@ -255,6 +255,218 @@ impl Component {
         self.output_edges = Some(o_edges);
         self.input_edges = Some(i_edges);
     }
+
+    /// Function for reducing the clocks found on the component.
+    /// Unused clocks and "duplicate" clocks (clocks that are never reset)
+    /// and then remove them.
+    pub fn reduce_clocks(&mut self, redundant_clocks: &Vec<RedundantClock>) {
+        for clock in redundant_clocks {
+            match &clock.reason {
+                ClockReductionReason::Duplicate(global) => self.replace_clock(&clock, global),
```

Comment:
You match clock as ref, so clippy is saying you can remove the ref from the call. Please change

Reply:
You are viewing an old commit, it should be fixed